internal/docker/handler_ssh.go

```diff
@@ -25,11 +34,146 @@ func (s *sshConnectionHandler) OnSessionChannel(
 	failureReason sshserver.ChannelRejection,
 ) {
 	return &channelHandler{
-		channelID:      channelID,
-		networkHandler: s.networkHandler,
-		username:       s.username,
-		exitSent:       false,
-		env:            map[string]string{},
-		session:        session,
+		channelID:         channelID,
+		networkHandler:    s.networkHandler,
+		connectionHandler: s,
+		username:          s.username,
+		exitSent:          false,
+		env:               map[string]string{},
+		session:           session,
 	}, nil
 }
+
+func (s *sshConnectionHandler) OnTCPForwardChannel(
+	channelID uint64,
+	hostToConnect string,
+	portToConnect uint32,
+	originatorHost string,
+	originatorPort uint32,
+) (channel sshserver.ForwardChannel, failureReason sshserver.ChannelRejection) {
+	channel, err := s.agentForward.NewForwardTCP(
+		s.setupAgent,
+		s.networkHandler.logger,
+		hostToConnect,
+		portToConnect,
+		originatorHost,
+		originatorPort,
+	)
+	if err != nil {
+		return nil, sshserver.NewChannelRejection(ssh.ConnectionFailed, message.EDockerForwardingFailed, "Error setting up the forwarding", "Error setting up the forwarding")
+	}
+	return channel, nil
+}
+
+func (s *sshConnectionHandler) OnRequestTCPReverseForward(
+	bindHost string,
+	bindPort uint32,
+	reverseHandler sshserver.ReverseForward,
+) error {
+	return s.agentForward.NewTCPReverseForwarding(
+		s.setupAgent,
+		s.networkHandler.logger,
+		bindHost,
+		bindPort,
+		reverseHandler,
+	)
+}
+
+func (s *sshConnectionHandler) OnRequestCancelTCPReverseForward(
+	bindHost string,
+	bindPort uint32,
+) error {
+	return s.agentForward.CancelTCPForwarding(bindHost, bindPort)
+}
+
+func (s *sshConnectionHandler) OnDirectStreamLocal(
+	channelID uint64,
+	path string,
+) (channel sshserver.ForwardChannel, failureReason sshserver.ChannelRejection) {
+	channel, err := s.agentForward.NewForwardUnix(
+		s.setupAgent,
+		s.networkHandler.logger,
+		path,
+	)
+	if err != nil {
+		return nil, sshserver.NewChannelRejection(ssh.ConnectionFailed, message.EDockerForwardingFailed, "Error setting up the forwarding", "Error setting up the forwarding (%w)", err)
+	}
+	return channel, nil
+}
+
+func (s *sshConnectionHandler) OnRequestStreamLocal(
+	path string,
+	reverseHandler sshserver.ReverseForward,
+) error {
+	return s.agentForward.NewUnixReverseForwarding(
+		s.setupAgent,
+		s.networkHandler.logger,
+		path,
+		reverseHandler,
+	)
+}
+
+func (s *sshConnectionHandler) OnRequestCancelStreamLocal(
+	path string,
+) error {
+	return s.agentForward.CancelStreamLocalForwarding(path)
+}
+
+func (c *sshConnectionHandler) setupAgent() (io.Reader, io.Writer, error) {
+	ctx, cancelFunc := context.WithTimeout(
+		context.Background(),
+		c.networkHandler.config.Timeouts.CommandStart,
+	)
+	defer cancelFunc()
+
+	if c.networkHandler.config.Execution.Mode == config.DockerExecutionModeConnection {
+		agent := []string{"/usr/bin/containerssh-agent", "forward-server"}
```

Comment:
This needs to be configurable. We already have config options for the agent path, maybe we can reuse that?